src/connectors/hierarchical-menu/connectHierarchicalMenu.js

```diff
@@ -186,13 +232,35 @@ export default function connectHierarchicalMenu(renderFn, unmountFn) {
           );
         }
 
+        const maxValuesPerFacetConfig = state.getQueryParameter(
+          'maxValuesPerFacet'
+        );
+        const currentLimit = this.getLimit();
+        // If the limit is the max number of facet retrieved it is impossible to know
+        // if the facets are exhaustive. The only moment we are sure it is exhaustive
+        // is when it is strictly under the number requested unless we know that another
+        // widget has requested more values (maxValuesPerFacet > getLimit()).
+        // Because this is used for making the search of facets unable or not, it is important
+        // to be conservative here.
+        const hasExhaustiveItems =
+          maxValuesPerFacetConfig > currentLimit
+            ? facetValues.length <= currentLimit
+            : facetValues.length < currentLimit;
```

Comment:
The engine doesn't handle this case for us?

Reply:
I'm not sure to understand which part could be handled by the engine here. Can you tell me more? I might have missed something.

Reply:
Yep me too, I messed up!